[IGNORE] Add upgrade guide about sql default value (#4294)
perses/perses: docs/upgrade-guide.md

### Upgrading from v0.53.0 to v0.54.0

#### SQL Database default configuration changes

> The following SQL database config field default value changed to be aligned with driver: https://github.com/go-sql-driver/mysql/blob/master/dsn.go#L97

Reviewer: ```suggestion
We have changed the behaviour of the SQL database config regarding the default value of few fields. It has been done to follow the default behaviour of the driver: https://github.com/go-sql-driver/mysql/blob/master/dsn.go#L97

If you do not use the SQL config, then this does not concern you.

If you use the SQL config, here the fields that the default value have been changed:
```

> allow_native_passwords: falseto true

Reviewer 2: ```suggestion
allow_native_passwords: false to true
```

> allow_native_passwords: falseto true

Reviewer: ```yaml
database:
  sql:
    allow_native_passwords: false to true

    check_conn_liveness: false to true
```

> check_conn_liveness: falseto true

Reviewer 2: ```suggestion
check_conn_liveness: false to true
```

> check_conn_liveness: falseto true

Reviewer: We also have added more field to the configuration to give you more facilities to customise the database connection:

```yaml
database:
  sql:
    # Maximum amount of time a connection may be reused. Keep it shorter than the server's wait_timeout
    # to avoid reusing connections the server has already closed.
    conn_max_lifetime: <duration> | default = 3m # Optional

    # Maximum amount of time a connection may be idle before it is closed.
    conn_max_idle_time: <duration> | default = 1m # Optional

    # Maximum number of open connections to the database. A value <= 0 means unlimited.
    max_open_conns: <int> # Optional

    # Maximum number of connections in the idle connection pool. A value <= 0 keeps the Go default (2).
    max_idle_conns: <int> # Optional
